--- /home/garming/.local/lib/python3.10/site-packages/xmlschema/validators/simple_types.py.bak	2024-08-15 09:04:54.536269587 +0000
+++ /home/garming/.local/lib/python3.10/site-packages/xmlschema/validators/simple_types.py	2024-08-15 09:11:03.389578935 +0000
@@ -25,9 +25,9 @@
     XSD_LENGTH, XSD_MIN_LENGTH, XSD_MAX_LENGTH, XSD_WHITE_SPACE, XSD_ENUMERATION, \
     XSD_LIST, XSD_ANY_SIMPLE_TYPE, XSD_UNION, XSD_RESTRICTION, XSD_ANNOTATION, \
     XSD_ASSERTION, XSD_ID, XSD_IDREF, XSD_FRACTION_DIGITS, XSD_TOTAL_DIGITS, \
-    XSD_EXPLICIT_TIMEZONE, XSD_ERROR, XSD_ASSERT, XSD_QNAME
+    XSD_EXPLICIT_TIMEZONE, XSD_ERROR, XSD_ASSERT, XSD_QNAME, XSD_NOTATION
 from ..translation import gettext as _
-from ..helpers import local_name
+from ..helpers import local_name, get_extended_qname
 
 from .exceptions import XMLSchemaValidationError, XMLSchemaEncodeError, \
     XMLSchemaDecodeError, XMLSchemaParseError
@@ -1470,9 +1470,14 @@
                         except (ValueError, DecimalException, TypeError):
                             pass
 
+                    if self.root_type.name in (XSD_QNAME, XSD_NOTATION):
+                        value = get_extended_qname(obj, kwargs.get('namespaces'))
+                    else:
+                        value = obj
+
                     for validator in self.validators:
                         try:
-                            validator(obj)
+                            validator(value)
                         except XMLSchemaValidationError as err:
                             yield err
 
